ocaml/rrdd/rrdd_main.ml

```diff
@@ -153,9 +153,13 @@ module Meminfo = struct
 				let meminfo_free = Int64.of_string (xs.Xs.read path) in
 				debug "memfree has changed to %Ld in domain %d" meminfo_free d;
 				current_meminfofree_values := IntMap.add d meminfo_free !current_meminfofree_values
-			with Xenbus.Xb.Noent ->
+			with
+			  Xenbus.Xb.Noent ->
 				debug "Couldn't read path %s; forgetting last known memfree value for domain %d" path d;
 				current_meminfofree_values := IntMap.remove d !current_meminfofree_values
+			| Failure("int_of_string") ->
```

Comment:
To avoid surprises, wouldn't it be a good idea to match all other exceptions here as well, to remove the current value, and to log it? I can't see what other exceptions could come up but I would argue that any exception means that the value was not read successfully.